src/Couchbase.Lite.Shared/API/Document/Blob.cs

```diff
@@ -280,6 +282,47 @@ internal Blob([NotNull]Database db, [NotNull]IDictionary<string, object> propert
             }
         }
 
+        internal Blob([NotNull]IDictionary<string, object> properties)
+        {
+            if (!Blob.IsBlob(properties)) {
+                throw new ArgumentException(CouchbaseLiteErrorMessage.InvalidJSONDictionaryForBlob);
+            }
+
+            Length = properties.GetCast<int>(LengthKey);
+            Digest = properties.GetCast<string>(DigestKey);
+            _properties = new Dictionary<string, object>(CBDebug.MustNotBeNull(WriteLog.To.Database, Tag,
+                nameof(properties), properties));
+            ContentType = properties.GetCast<string>(ContentTypeKey);
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Return whether the given dictionary represents Blob or not
+        /// </summary>
+        /// <param name="blobDict"> JSON Dictionary represents in the <see cref="Blob"/> :
+        /// Key          | Value                  | Mandatory | Description
+        /// ---------------------------------------------------------------------------------------------------
+        /// @type        | constant string “blob” | Yes       | Indicate Blob data type.
+        /// content_type | String                 | No        | Content type ex. text/plain.
+        /// length       | Number                 | No        | Binary length of the Blob in bytes.
+        /// digest       | String                 | Yes       | The cryptographic digest of the Blob’s content.
+        /// </param>
+        /// <returns>Return true if the given dictionary represents Blob, otherwise return false</returns>
+        public static bool IsBlob(IDictionary<string, object> blobDict)
+        {
+            if (!blobDict.ContainsKey(Constants.ObjectTypeProperty) || (string) blobDict[Constants.ObjectTypeProperty] != Constants.ObjectTypeBlob
```

Comment:
Will throw an invalid cast exception if it is not a string.

Reply:
Add an additional check if blobDict[Constants.ObjectTypeProperty] is string before cast it.

Reply:
This will do, `as` could have compared them in one line though